### PR TITLE
feat: Upgrade RPC methods

### PR DESCRIPTION
**Pull request by andyv09**
Update RPC methods according to internal docs

### CODE REVIEW

**Review by Vid201 — DISMISSED** [review]

File: packages/snap/src/utils/veramoUtils.ts [hunk 1]
```diff
@@ -52,12 +56,34 @@ export async function veramoListVCs(
...
+  };
+};
 export async function veramoCreateVP(
   params: ApiParams,
-  vcId: string,
-  challenge?: string,
-  domain?: string
+  createVPParams: CreateVPRequestParams
 ): Promise<VerifiablePresentation | null> {
+  const id = createVPParams.vcs[0].id;
```
Comment (martines3000) [inline packages/snap/src/utils/veramoUtils.ts:81]: Shouldn't we use all the VCs ? Or is it maybe not supported by Veramo yet ?

Comment (martines3000) [reply packages/snap/src/utils/veramoUtils.ts:81]: We can also do this in the next PR, but it needs to be the top priority.

File: packages/snap/src/utils/veramoUtils.ts [hunk 2]
```diff
@@ -52,12 +56,34 @@ export async function veramoListVCs(
...
+  const store = createVPParams.vcs[0].metadata?.store;
```
Comment (martines3000) [inline packages/snap/src/utils/veramoUtils.ts:82]: This is unused and we need to find a better way to handle the querying on different store &rarr; the current try catch approach is not really scalable (and it looks ugly :wink:)

Comment (martines3000) [reply packages/snap/src/utils/veramoUtils.ts:82]: But can be done together with the above comment in the next PR

File: packages/snap/src/utils/veramoUtils.ts [hunk 3]
```diff
@@ -81,6 +107,7 @@ export async function veramoCreateVP(
     const promptObj = {
       prompt: 'Alert',
       description: 'Do you wish to create a VP from the following VC?',
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
```
Comment (martines3000) [inline packages/snap/src/utils/veramoUtils.ts:110]: I don't think this is needed

File: packages/snap/src/rpc/did/switchMethod.ts [hunk 4]
```diff
@@ -1,33 +1,28 @@
...
 import { ApiParams } from '../../interfaces';
 import { changeCurrentMethod } from '../../utils/didUtils';
 import { snapConfirm } from '../../utils/snapUtils';
 
 export async function switchMethod(
   params: ApiParams,
-  didMethod: string
-): Promise<boolean> {
+  didMethod: AvailableMethods
+): Promise<string> {
```
[the remark this answers is not in the document]
Comment (martines3000) [reply packages/snap/src/rpc/did/switchMethod.ts:9]: Maybe we can leave it as string and change it in one of the next PRs where we add proper returns (error, results,...)

**Review by martines3000 — APPROVED** [review]
Looks good now.

**Pull request merged**